include/cuco/detail/trie/bit_vector/bit_vector.cuh

```diff
@@ -0,0 +1,200 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuco/detail/trie/bit_vector/bit_vector_ref.cuh>
+#include <cuco/extent.cuh>
+#include <cuco/storage.cuh>
+#include <cuco/utility/allocator.hpp>
+
+#include <thrust/device_vector.h>
+
+#include <cuda/atomic>
+
+namespace cuco {
+namespace experimental {
+
+/**
+ * @brief Struct to store ranks of bits at 256-bit intervals
```

Comment:
Is 256 chosen for performance purposes?

Reply:
It is a tradeoff between space efficiency and performance. Larger values will create a smaller index, but searches will be slower.

This is hardcoded to 256 (four 64-bit words) now, but can be made configurable. Main change would be in `rank` structure which stores a single base and deltas for each 64-bit word.